src/client.ts

```diff
@@ -483,19 +645,127 @@ export class MultiServerMCPClient {
     };
   }
 
+  private _getHttpErrorCode(error: unknown): number | undefined {
+    const streamableError = error as StreamableHTTPError | SseError;
+    let { code } = streamableError;
+    // try parsing from error message if code is not set
+    if (code == null) {
+      const m = streamableError.message.match(/\(HTTP (\d\d\d)\)/);
+      if (m && m.length > 1) {
+        code = parseInt(m[1], 10);
+      }
+    }
```

Comment:
This is probably the grossest part - I needed this during early testing because for some reason the `code` property of the error wasn't being set. It appears that it is being set now, but I decided to leave it in for reliability.